### PR TITLE
Update compatibility docs

### LINKED ISSUE #2359: Official Sinon docs are erroneous with regards to browser support
**Describe the bug**
See #2358 for background. The documentation at https://sinonjs.org/releases/latest/ mentions Internet Explorer 11 and many others as supported runtimes. This is wrong. We should probably just remove the entire text block and describe our official stance for the project and point to the browsersupport and/or test suite for saucelabs.

### PR DESCRIPTION
**Pull request by fatso83**
#### Purpose (TL;DR) - mandatory

Close #2359 by updating the docs to reflect our current stance.

 #### Background (Problem in detail)  - optional

Our docs state that we support Safari 9 and IE 11 - which we do not. This creates bugs such as #2358.

 #### Solution  
Point to the up-to-date reference and talk in general terms about our strategy.

#### How to verify - mandatory

1. Check out this branch
2. Follow steps in our docs repo to see local version.

### CODE REVIEW

**Review by mroderick — APPROVED** [review]
Very nice 👍

**Comment by mroderick** [comment]
You might also want to update `README.md`

**Comment by fatso83** [comment]
@mroderick I got some new questions after inspecting the README and the linked COMPATIBILITY.md file. 

The compat file touts itself as being the canonical reference. This prompts some questions. 
1. Maybe it is wrong to point to the eslint-config and I should just point this compat file instead?
2. How can we keep the docs site, the the compatibility file and the plugin in sync? Copy-on-install?
3. I see the compat document state that we use [eslint-plugin-compat](https://www.npmjs.com/package/eslint-plugin-compat), but I do not see it as a direct dependency, although it is in the lock file. It is also not mentioned in our config, nor the shared config. How is this being enforced?

**Comment by mroderick** [comment]
> 1. Maybe it is wrong to point to the eslint-config and I should just point this compat file instead?

I agree, that's a better approach.

> 2\. How can we keep the docs site, the the compatibility file and the plugin in sync? Copy-on-install?

I have not figured that out yet.

What do you mean by copy-on-install? Maybe that's the solution?

> 3\. I see the compat document state that we use [eslint-plugin-compat](https://www.npmjs.com/package/eslint-plugin-compat), but I do not see it as a direct dependency, although it is in the lock file. It is also not mentioned in our config, nor the shared config. How is this being enforced?

It is installed in consuming projects as [a dependency of `@sinonjs/eslint-config`](https://github.com/sinonjs/eslint-config/blob/6dc048ac261f7aa7a57129314278247196499bcd/package.json#L25)

**Comment by fatso83** [comment]
Mange tak, Morgan!

> What do you mean by copy-on-install? Maybe that's the solution?

Since the browserlist file is _text_ and in our code base, and our docs are also in the same code base, then we could use that file to update other documents by running some `post-build` script when doing `npm run build` or something. Just loose thoughts. Something like
```
<!-- browserlist start -->
blabla, contents of .browserlist from eslint-config
<-- browserlist end-->
```
and replace the bits between the magic start and end comments each time.

For now, I guess it's much easier to talk in general terms and point to some code for the most specific up-to-date details.

**Comment by mroderick** [comment]
> Since the browserlist file is _text_ and in our code base, and our docs are also in the same code base, then we could use that file to update other documents by running some `post-build` script when doing `npm run build` or something. Just loose thoughts. Something like
> 
> ```
> 
> blabla, contents of .browserlist from eslint-config
> <-- browserlist end-->
> ```
> 
> and replace the bits between the magic start and end comments each time.

For the markdown documents, I think that's a great idea.
For the `.browserslistrc` file, we can replace it outright.

> For now, I guess it's much easier to talk in general terms and point to some code for the most specific up-to-date details.

I agree, that's a good approach 👍

**Comment by fatso83** [comment]
https://github.com/sinonjs/eslint-config/pull/8

**Comment by fatso83** [comment]
If you could have a new look, I think I have removed most ambiguity by pointing to a single doc that is kept up-to-date by a script.

**Review by mroderick — APPROVED** [review]
This looks good 👍 

I tried running `npm run build` locally, which didn't fail at least :)

File: scripts/update-compatibility.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+debug(
+    `Inline browserslistrc file into ${compatibilityPath} to keep contents in-sync`
+);
+const rl = readline.createInterface({
+    input: fs.createReadStream(compatibilityPath),
+});
+
+let changeLogData = "";
+
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (fatso83) [reply scripts/update-compatibility.js:32]: I started just disabling the jsdoc rule, then I needed to disable another one, and then I was like: "screw it, this is just a util", but sure, I could list them.

**Pull request merged**